### PR TITLE
Add xml comments to field resolvers

### PR DESCRIPTION
**Pull request by Shane32**
See changed files

### CODE REVIEW

File: src/GraphQL/Resolvers/AsyncFieldResolver.cs [hunk 1]
```diff
@@ -24,6 +28,7 @@ object IFieldResolver.Resolve(IResolveFieldContext context)
         }
     }
 
+    /// <inheritdoc cref="AsyncFieldResolver{TReturnType}"/>
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/AsyncFieldResolver.cs]: Would you like to say a few words about how this resolver differs from the previous one?

Comment (Shane32) [reply src/GraphQL/Resolvers/AsyncFieldResolver.cs]: Sure

File: src/GraphQL/Resolvers/FuncFieldResolver.cs [hunk 2]
```diff
@@ -23,6 +26,7 @@ object IFieldResolver.Resolve(IResolveFieldContext context)
         }
     }
 
+    /// <inheritdoc cref="FuncFieldResolver{TReturnType}"/>
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/FuncFieldResolver.cs]: Here the same.

**Comment by Shane32** [comment]
By the way, I think I’m about done with xml comments. I think the FieldType could really use some xml comments, probably some of the IGraphType and related interfaces, perhaps IValidationRule, and personally I’d do all the field builder methods. But I’m running out of ideas and have some other things I’d like to spend time on. If you want to take a stab at any of the above I’ll be happy to work with you on it.

File: src/GraphQL/Resolvers/NameFieldResolver.cs [hunk 3]
```diff
@@ -5,6 +5,9 @@
 
 namespace GraphQL.Resolvers
 {
+    /// <summary>
+    /// Attempts to return a value for a field from the graph's source object, matching the name of the field to a property on the source object
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/NameFieldResolver.cs]:
```suggestion
    /// Attempts to return a value for a field from the graph's source object, matching the name of the field to a property or a method with the same name on the source object
```

File: src/GraphQL/Resolvers/IFieldResolver.cs [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+    /// <see cref="ComplexGraphType{TSourceType}.Field{TProperty}(System.Linq.Expressions.Expression{System.Func{TSourceType, TProperty}}, bool, System.Type)">Field(x => x.Name)</see>,
+    /// <see cref="ExpressionFieldResolver{TSourceType, TProperty}">ExpressionFieldResolver</see> is used.<br/>
+    /// <br/>
+    /// When a field resolver is not defined, such as with <see cref="ComplexGraphType{TSourceType}.Field{TGraphType, TReturnType}(string)">Field("Name")</see>,
+    /// the static instance of <see cref="NameFieldResolver"/> is used.
+    /// </summary>
     public interface IFieldResolver
     {
+        /// <summary>
+        /// Returns an object, null, or a <see cref="Task{TResult}"/> for the specified field
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/IFieldResolver.cs]:
```suggestion
        /// Returns an object, null, or a <see cref="Task{TResult}"/> for the specified field. If the task is returned, then this task will be awaited to obtain the actual object.
```

**Comment by sungam3r** [comment]
>  and have some other things I’d like to spend time on

So do I.

**Pull request merged**